### PR TITLE
ARROW-1950: [Python] pandas_type in pandas metadata incorrect for List types

### PR DESCRIPTION
**Pull request by cpcloud**

### CODE REVIEW

**Comment by cpcloud** [comment]
I'm going to add a C++ test for empty list typed arrays as well.

File: python/pyarrow/tests/test_convert_pandas.py [hunk 1]
```diff
@@ -1404,6 +1404,57 @@ def test_empty_list_roundtrip(self):
 
         tm.assert_frame_equal(result, df)
 
+    def test_empty_list_metadata(self):
+        # Create table with array of empty lists, forced to have type
+        # list(string) in pyarrow
+        c1 = [["test"], ["a", "b"], None]
+        c2 = [[], [], []]
+        arrays = {
```
Comment (xhochy) [inline python/pyarrow/tests/test_convert_pandas.py]: You will need to use `OrderedDict` here probably to get the same ordering of the results in all Python versions.

Comment (cpcloud) [reply python/pyarrow/tests/test_convert_pandas.py]: Yep, thanks! I saw that on the CI :)

**Comment by cpcloud** [comment]
@xhochy do you mind if i merge this one when it passes? i want to see if my gitbox powers are working.

**Review by xhochy — APPROVED** [review]
@cpcloud yes. This looks good

**Comment by cpcloud** [comment]
Passing appveyor build here: https://ci.appveyor.com/project/cpcloud/arrow/build/1.0.520

**Comment by cpcloud** [comment]
Sweet.

**Pull request closed without merging**